src/cloud_run/tables/dim_tollway.py

```diff
@@ -1,48 +1,17 @@
-from google.api_core.exceptions import GoogleAPIError
-
-from utils.table_logger import setup_logger
-from utils.constants import DATASET_ID, PROJECT_ID, TABLES
+from utils.constants import TABLES
 from utils.helpers import hash_string
-
-
-def get_tollway_names(bigquery_client):
-    table = f"{PROJECT_ID}.{DATASET_ID}.{TABLES['dim_tollway']}"
-    query = f"""
-        SELECT DISTINCT tollway_name
-        FROM `{table}`
-    """
-    tollway_names = bigquery_client.query(query).result()
-    return [row["tollway_name"] for row in tollway_names]
+from tables.table_manager import TableManager
 
 
 def insert_row(bigquery_client, message_data):
 
-    tollway_names = get_tollway_names(bigquery_client)
-    tollway_logger = setup_logger(True)
-
-    if message_data.tollway_name in tollway_names:
-        tollway_logger.info(f"{message_data.tollway_name} exists in dim_tollway")
-        return
-
+    tollway_id = hash_string(message_data.tollway_name)
     dim_tollway_row = {
-        "tollway_id": hash_string(message_data.tollway_name),
+        "tollway_id": tollway_id,
         "tollway_name": message_data.tollway_name,
     }
 
-    table_ref = bigquery_client.dataset(DATASET_ID).table(TABLES["dim_tollway"])
-
-    try:
-        dim_tollway_errors = bigquery_client.insert_rows_json(table_ref, [dim_tollway_row])
-        tollway_logger.info(f"Successfully inserted {message_data.tollway_name} into BigQuery")
-    except GoogleAPIError as ge:
-        tollway_logger.error(
-            "Encountered errors while inserting tollway " f"{message_data.tollway_name} into BigQuery: {ge}"
-        )
-        raise
-    except Exception as e:
-        tollway_logger.error(
-            "An unexpected error occurred while inserting tollway "
-            f"{message_data.tollway_name} into BigQuery: {e}",
-            exc_info=True,
-        )
-        raise
+    table_manager = TableManager(client=bigquery_client, table=TABLES["dim_tollway"], row=dim_tollway_row)
```

Comment:
this and next module's code seems similar to the previous one?

Reply:
Sounds good, sometimes we have to in order to figure things out, but now that the duplication is there you can refactor and use the extract method refactoring.